Religion fixups

LT-Rascek opened the pull request:
The umpteenth Religion fixups branch. Merged in Dharmachakra as part of BP; we need be certain to put a thanks to herkles at or near the top since it helped greatly in India and provided a number of good graphics.

There's a number of bugs still floating around relating to the Dharmachakra merge that I need to squash. I also want to tweak some of Dharmachakra's setup to use the Religion Doctrines Pane instead of listing them as special doctrines.

Includes a number of tweaks to how ecumenism in Christianity works. I probably want to do a deeper dive into a few tenets for Christians as well as seeing if there can be some more improvements.

LT-Rascek commented:
* The new GUI fixes now only display religion doctrines if the faith actually has some doctrines.
* Hinduism now has doctrines relating to how they view the supreme deity organized similarly to how Christian Doctrines are organized
* A large number of tweaks to various religious features.

LT-Rascek commented:
### Remaining To Do list on this PR:
* Implement Hindu subsects as decision (in progress)
* Monument building decision like raising a runestone for Germanics

* Fix Dhamchakra-porting error list
	* Need to fix missing pilgrimage localization bugs
	* Fix unrecognized localizations about pet elephants, sallekhana (consolatium)
	* Fix Dharmachakra's tibetan diety refences for buddhism flavor events
	* Fix elephant story circle issues
	* Fix never_nudity references

* Remove Nuristani religion after merge

### Stretch Goals for this PR (not necessary but would be nice):
* See if new category besides between shunned and accepted is possible for antinomians (shunned effects w/o hooks or devotion loss, just opinion loss)
	* Yes, but touches quite a bit of code for a more precise flavorization and opinion modeling. Not impossible, but not simple.

* Revisit Christian doctrine and tenet interactions
	* Demigurge: No wandering Jews in your court
	* Judaizers: Increased chances for talented Jews in your court

loup99 commented on BP/common/religion/religions/00_baltic.txt:
```diff
@@ -194,6 +194,7 @@
 	}
 
 	faiths = {
+		#WtWSMS Differences: No raiding, **can** declare HW
```
I'm not convinced on these differences.

LT-Rascek replied:
Raiding is tied unreformed pagan, tribal government, or certain dynasty modifiers (not something implicit in a religion) and HW casus belli requires organization and non-pacficism. So a reformed Baltic won't be able to raid, but will be able to HW.

LT-Rascek commented:
Summary of the state of the branch at the latest commit:

New Tenets:
* Peregrinition tenet added; combines asceticism and monasticism along with a peregrinition decision.
* Ethnic religion tenet added; applies to various minority faiths that survived for a long time, some well into the modern day (bon_religion, tani_religion, kalashism, mandeanism, most jewish faiths, vasconic_pagan, basque_pagan). Not available to universalist faiths (Christian, Islam, Zoroastrian, Buddhists). Strong county conversion resistance, improved conversion speed in same county culture, strongly reduced conversion speed in counties of a different culture, occasional religion reversions, and improved same-faith opinion.
* Apostolic Succession replaces Universal Church; same effects as communion and is not selectable at religion creation but (in beta) should be tied to council-type mechanics.
* Autocephaly tenet added for some Christian faiths; similar to communal identity _but_ it applies to a whole culture group, provides no promote culture benefit, and has some conversion resistance. Reason being is autocephalous churches seemed to be applied (in Vanilla CK3) at a culture group level and autocephalous churches seemed to last pretty long (e.g., Armenia, Syria) in the face of Muslim conquest. Beta plan is to add an option for players to create their own autocephalous church if one doesn't exist for their culture group and chosen Christology Creed.
* Dhammavicaya tenet added for Buddhsists; similar to Literalism but also has a monthly lifestyle learning bonus [Dharmachakra].

Removed Tenets:
* Upland Isolation; better represented by ethnic religion and the name isn't tied to a specific terrain type.

Changed Tenets:
* Rigorism properly considers all other Christian faiths evil again.
* Pastoral Isolation no longer considers all other Christians righetous, as the new ecumenism doctrines handle those cases.
* Divine Marriage is dynamically renamed Xwedodah for Zoroastrian & Hepthal religions, and has special descriptions for Christians, Dualists, Jews, and Muslims as well as Zoroastrians and Hepthals.

Doctrine Changes:
* Added Chalcedeon Creed and Miaphysite Communion main doctrines to apply to their respective faiths. Both doctrines consider themselves righteous and the other astray.
  * Chalcedeon Creed:  Nicene, Georgian Orthodox, Insular Celtic
  * Miaphysite Communion: Miaphysitism, Armenian Apostolic, Syriac Orhtodox
* Cleaned up Christian doctrine combination possibilities
* Refactored Christian doctrine costs depending on gameplay effect.
* Soteriology now adds traits and virtues depending on choice, instead of just antinomianism.
* Sampradaya now converted to a Hindu religion Doctrine instead of a main doctrine [Dharmachakra].
* Sacred Eunuchs (Cybelian) now makes all clergy eunuchs and monks.

New Faiths:
* Syrian Orthodox: Communion, Monasticism, Autocephalous Church (Christian)
* Armenian Apostolic: Communion, Monasticism, Autocephalous Church (Christian)
* Kalashism: Bhakti, Ritual Hospitality (Puja), Ethnic Religion [unreformed faith, Shaivism Sampradaya] (Hindu); represents the proto-Vedic faith of the Nuristanti and Kalash people. Has unique holy sites.
* Eiateuta (Illyrian Pagan): Ancestor Worship, Sanctity of Nature, Ethnic Religion [unreformed faith]. Uses WtWSMS holy sites. Imported because #54 has some provinces to be Illyrian Pagan for reasons discussed in that branch.

Revised Faith Tenets:
* Christian Faiths:
  * Kartevalian Orthodoxy: Communion, Monasticism, Autocephalous Church
  * Nestorian: Monasticism, Adaptive, Apostolic Succession
  * Nicene: Monasticism, Pentarchy, Apostolic Succession
  * Miaphysitism: Alexandrian Catechism, Apostolic Succession, Peregrinition
  * Insular Celtic: Vows of Poverty, Pastoral Isolation, Peregrinition
  * Pelagianism: Armed Pilgrimages, Monasticism, Vows of Poverty
  * Audianism: Aniconism, Alexandrian Catechism, and Literalism (because of Anthropomorphism)
* Jewish Faiths:
  * Haymanot: Monasticism, Religious Law [Halakha], and Ethnic Religion
  * Samaritanism: Esotericism, Literalism, Ethnic Religion
  * Kararism: Literalism, Religious Law [Halakha], Ethnic Religion
  * Tzadikite: Unrelenting Faith, Halakha, Ethnic Religion
  * Malabarism: Religious Law [Halakha], Eastern Syncretism, Ethnic Religion
* Zoroastrian Faiths:
  * Arewordik: Unrelenting Faith, Sanctioned False Conversion, Hellenism Syncretism (more representative of the fact that Armenia was a fusion of Persian and Roman influences)
* Early Buddhist Schools:
  * Dharmaguptaka uses Communal Identity, Dharmic Pacifism, and Inner Journey
  * Sarvastivadan uses Adaptive, Dharmic Pacifism, and Dhamm Vicaya.
  * Mahasamghika uses Esotericism, Dharmic Pacifism, and Inner Journey
  * Pudgalavada uses Religious Law, Dharmic Pacifism, and Dhamma Vicaya.
* Bon:
  * Khyarwe Bon: Patron Gods, Gruesome Festivals/Human Sacrifice, Ethnic Religion
  * Gyur Bon: Monasticism, Inner Journey, Ethnic Religion
 * Tani Faiths:
   * Donyi-Poloism: Religious Law, Ancestor Worship, Ethnic Religion
   * Sedism: Religious Law, Sanctity of Nature, Ethnic Religion
   * Bathouism: Inner Journey, Sanctity of Nature, Ethnic Religion
   * Sanamahism: Ancestor Worship, Sanctity of Nature, Ethnic Religion
 * Mundhumism:
   * Kiratism: Reincarnation, Ritual Celebrations, Ethnic Religion
   * Ymaism: Ritual Celebrations, Sacred Childbirth, Ethnic Religion
  * Dualism:
    * Mandeanism: Esotericism, Astrology, and Ethnic Religion
  * Pagan:
    * Armazism: Ancestor Worship, Sanctity of Nature, Ethnic Religion
    * Vasconic: Ancestor Worship, Sanctity of Nature, Ethnic Religion
    * Zalmoxian: Ancestor Worship, Warmonger, Hellenism Syncretism
    * Sarnaism: Ancestor Worship, Sanctity of Nature, Ethnic Religion

New Faith Symbols:
Patripassianism & Montanism & Donatism (@Abian36), Kartevalian Orthodoxy, Doceticism, Insular Celtic, Pelagianism, Miaphysitism, Aquelianism, Amazighism, Syriac Orthodoxy, Hepthalitism, Kalashism, Bathouism, Sanhamhism, Priscillianism, Valentinianism, Sethianism, Mandeanism, Sabianism, Caintisim, Platonism, Germanic Paganism

Other Faith Changes:
* The Pope in Rome is head of the Nicene Church like WtWSMS.
* All Tani faiths use the same holy sites.
* Jewish faiths are polygamous, as they were historically; over time, as a minority religion, Jews adopted the marriage practices of the locals and became mostly monogamous.
* Slavic pagan is the same color it was in WtWSMS
* Hindus can adopt one of 6 philosophical schools if they have the scholar trait.

Other Changes:
* Some GUI rework makes it so religions without a religion doctrine don't have that portion show up in the GUI.

Known To-Dos:
* Replace Bon patron gods with the ones from Dharmachakra
* Add new localizations for patron gods tenet for the Kalash religion and Dravidian cultures.

loup99 commented on BP/common/religion/_ToDo.info:
```diff
@@ -0,0 +1,60 @@
+To Do Tasks:
+Fix some remaining errors:
+	[12:18:33][jomini_dynamicdescription.cpp:66]: Unrecognized loc key special_doctrine_christian_ecumenism_name. special_doctrine_christian_ecumenism
```
Do you plan on doing this before or after merge?

LT-Rascek replied:
Fixed already; forgot to remove this in the to-dos file.

loup99 commented on BP/common/religion/_ToDo.info:
```diff
@@ -0,0 +1,60 @@
+==================
+
+Following WtWSMS faiths are still not included:
+Eiateuta (Illyrian Pagan), Hubalism, Wodenism
```
Wodenism and Hubalism aren't faiths used in WtWSMS and you have imported Eiateuta. With this in mind the import is finished.

loup99 commented on BP/common/religion/_ToDo.info:
```diff
@@ -0,0 +1,60 @@
+Eiateuta (Illyrian Pagan), Hubalism, Wodenism
+
+Following WtWSMS faiths aren't included (and shouldn't be):
+Anything relating to Islam (for now), Aztecs, Hindu subsects (to be implemented as decisions)
```
The Aztecs are from CK2 vanilla, not the mod.

loup99 commented on BP/common/religion/doctrines/dharma_doctrines_hinduism.txt:
```diff
@@ -0,0 +1,109 @@
+#
+#Dharmachakra is a flavor mod for CK3 focused on the Indian Subcontient and the neighboring regions of Burma and Tibet, as well as the Dharmic Religions of Buddhism, Jainism, and Hinduism and religions influenced by them.
+#
+#Feel free to use my work in your mods - all I ask is that I am credited and (if possible) to give me a heads up.
```
Same as I said elsewhere, credits don't need to be included everywhere.

loup99 commented on BP/common/religion/religions/00_donyipoloism.txt:
```diff
@@ -1,17 +1,17 @@
-#into this family as well; they are all
-#Tibeto-Burman faiths
+﻿#LT-Rascek: Taken from herkles "Dharmachakra" and modified; moved bathouism and sanamahism into this family as well; 
+# they are all Tibeto-Burman faiths
```
Is Sanamahi really Tibeto-Burman?

LT-Rascek replied:
It's the ethnic religion of the Meitei people and the Meitei people are Tibeto-Burman.

loup99 replied:
Manipur is generally not considered Burman, hence my interrogation. It is a detail nonetheless.

LT-Rascek replied:
It's not Burman but Meitei is a Tibeto-Burman language. Tibeto-Burman, at least the way I put it together, is the catch-all for cultures that are not properly Tibetan nor properly Burman. So the Pyu and Meitei fall into that category.

loup99 replied:
Ok

loup99 commented on BP/common/religion/religions/00_hellenism.txt:
```diff
@@ -323,6 +324,7 @@
 				CreatorHerHim = CHARACTER_HERHIM_IT
 			}
 		}
+		#WtWSMS differences: no raiding, **can't** be reformed, has HOF
```
Reformable head of faith, but not preexisting?

LT-Rascek replied:
From WtWSMS Latin Solar:
* There is (and can exist) no religious head.
* Can be reformed
* Can raid infidel neighbors for loot

The Sol Invictus Cult, otoh, should be organized (like Hellenism itself) as it was an official imperial cult for a while. So there's no need to reform it (organized), and I should add a specific head of faith (temporal) instead of letting CK3 auto-generate it.

loup99 replied:
I disagree, by 476 there are no longer any institutions and it is no longer reorganised. Reformation is the process that allows it to be reinstituted. Thus it should be disorganised, reformable and without head. In WtWSMS Latin Solar by default had a temporal and secular head, a bit like the Norse.

LT-Rascek replied:
Alright, will make it unreformed.

loup99 commented on BP/common/religion/religions/00_zoroastrianism.txt:
```diff
@@ -468,6 +473,8 @@
 			doctrine = doctrine_spiritual_head
 		}
 		
+		#WtWSMS differences: has a HOF
```
For a post-Islam faith?

LT-Rascek replied:
That's a default CK3 thing.

loup99 replied:
Ok, but for the mod I believe it shouldn't be active at start and be without a head until an actor institutes it, after it has appeared.

LT-Rascek replied:
It's not active (i.e., on the map) at start. We'll need to add variables/flags to the game as well as manipulate files in common/scripted_character_templates/ to prevent pre-Islam religions from spawning; I've had a few Ashari wanderers before.

But those are for after this PR and after the map is finished (it's a feature to be implement for pre-alpha).

loup99 commented on BP/common/religion/religions/BP_berber.txt:
```diff
@@ -191,6 +191,7 @@
 	}
 
 	faiths = {
+		#WtWSMS: talk with Enlil
```
Are there any remaining questions other than what was discussed?

LT-Rascek replied:
Nope. Removed.

loup99 commented on BP/common/religion/religions/BP_bosporan.txt:
```diff
@@ -191,7 +191,7 @@
 
 	faiths = {
-		#WtWSMS differences: No learning bump, no opinion buff/nerf (+20/-10), no fertility boost
+		#WtWSMS differences: No learning bump, no fertility boost, **can** be reformed
```
In WtWSMS it is not a Pagan faith.

LT-Rascek replied:
Fixed.
